### PR TITLE
Added opt-out approach to Telemetry.

### PR DESCRIPTION
**Pull request by popovaan**
Added opt-out approach to Telemetry.
Ticket: 110429

### CODE REVIEW

File: src/main.py [hunk 1]
```diff
@@ -56,6 +64,17 @@ def __init__(self, app_name: str = None, app_version: str = None, tid: str = Non
         if self.consent and not self.backend.cid_file_initialized():
             self.backend.generate_new_cid_file()
 
+        if not enable_opt_in_dialog and self.consent:
+            # Try to create directory for client ID if it not exists
```
Comment (rkazants) [inline src/main.py:68]:
```suggestion
            # Try to create directory for client ID if it does not exist
```

File: src/main.py [hunk 2]
```diff
@@ -196,7 +215,7 @@ def _update_opt_in_status(tid: str, new_opt_in_status: bool):
         app_name = 'opt_in_out'
         app_version = Telemetry.get_version()
         opt_in_checker = OptInChecker()
-        opt_in_check = opt_in_checker.check()
+        opt_in_check = opt_in_checker.check(enable_opt_in_dialog=False)
```
Comment (rkazants) [inline src/main.py:218]: can you add a comment why it is switched off here?

**Review by rkazants — APPROVED** [review]

File: src/main.py [hunk 3]
```diff
@@ -34,17 +34,25 @@ class Telemetry(metaclass=SingletonMetaClass):
     application name, version and tracking id just once. Later the instance can be created without parameters.
     """
     def __init__(self, app_name: str = None, app_version: str = None, tid: str = None,
-                 backend: [str, None] = 'ga'):
+                 backend: [str, None] = 'ga', enable_opt_in_dialog=True):
```
Comment (rkazants) [inline src/main.py:37]: please add description for this new flag in the comments

Comment (popovaan) [reply src/main.py:47]: Done.

File: src/main.py (hunk 2 above)
Comment (popovaan) [reply src/main.py:233]: Added comment.

**Pull request merged**